galley/pkg/config/analysis/analyzers/namespaceconflict/peerauthentication.go

```diff
@@ -0,0 +1,112 @@
+package namespaceconflict
+
+import (
+	"fmt"
+
+	v1beta1 "istio.io/api/security/v1beta1"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+	k8s_labels "k8s.io/apimachinery/pkg/labels"
+)
+
+// Analyzer checks conditions related to conflicting namespace level resources
+type PeerAuthenticationConflictAnalyzer struct{}
+
+var _ analysis.Analyzer = &PeerAuthenticationConflictAnalyzer{}
+
+var peerAuthenticationCol = collections.IstioSecurityV1Beta1Peerauthentications.Name()
+
+// Metadata implements Analyzer
+func (a *PeerAuthenticationConflictAnalyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "namespaceconflict.PeerAuthenticationConflictAnalyzer",
+		Description: "Checks conditions related to Peer Authentication conflicting namespace level resources",
+		Inputs: collection.Names{
+			peerAuthenticationCol,
+		},
+	}
+}
+
+func (a *PeerAuthenticationConflictAnalyzer) Analyze(c analysis.Context) {
+	namespaceWideConfiguration := map[string]*resource.Instance{}
+
+	c.ForEach(peerAuthenticationCol, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.PeerAuthentication)
+		xNS := r.Metadata.FullName.Namespace.String()
+
+		// If the resource has workloads associated with it, analyze for conflicts with selector
+		if x.GetSelector() != nil {
```

Comment:
This code is common amongst all of these. is it possible to have a single analyzer that shares all of this? each type probably needs a few lines doing something like:
```
c.ForEach(<TYPE>, func() {
  checkConflicts(x.GetSelector())
})
```

or similar

Reply:
Hey @howardjohn, I was definitely searching for a way to have something similar to this. I'll add this capability and submit a new PR. Thanks a lot for pointing me in the right direction